src/coordinator/api/v1/handler/database/create.go

```diff
@@ -0,0 +1,249 @@
+func NewCreateHandler(client clusterclient.Client, cfg config.Configuration) http.Handler {
+	return &createHandler{
+		placementInitHandler:   placement.NewInitHandler(client, cfg),
+		namespaceAddHandler:    namespace.NewAddHandler(client),
+		namespaceDeleteHandler: namespace.NewDeleteHandler(client),
+	}
+}
+
+func (h *createHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := logging.WithContext(ctx)
+
+	namespaceRequest, placementRequest, rErr := h.parseRequest(r)
+	if rErr != nil {
+		logger.Error("unable to parse request", zap.Any("error", rErr))
+		handler.Error(w, rErr.Error(), rErr.Code())
+		return
+	}
+
+	nsRegistry, err := h.namespaceAddHandler.Add(namespaceRequest)
+	if err != nil {
+		logger.Error("unable to add namespace", zap.Any("error", err))
+		handler.Error(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	initPlacement, err := h.placementInitHandler.Init(placementRequest)
+	if err != nil {
+		// Attempt to delete the namespace that was just created to maintain idempotency
+		err = h.namespaceDeleteHandler.Delete(namespaceRequest.Name)
+		if err != nil {
+			logger.Error("unable to delete namespace we just added", zap.Any("error", err))
+			handler.Error(w, err, http.StatusInternalServerError)
+			return
+		}
+
+		logger.Error("unable to add namespace", zap.Any("error", err))
+		handler.Error(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	placementProto, err := initPlacement.Proto()
+	if err != nil {
+		logger.Error("unable to get placement protobuf", zap.Any("error", err))
+		handler.Error(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	resp := &admin.DatabaseCreateResponse{
+		Namespace: &admin.NamespaceGetResponse{
+			Registry: &nsRegistry,
+		},
+		Placement: &admin.PlacementGetResponse{
+			Placement: placementProto,
+		},
+	}
+
+	handler.WriteProtoMsgJSONResponse(w, resp, logger)
+}
+
+func (h *createHandler) parseRequest(r *http.Request) (*admin.NamespaceAddRequest, *admin.PlacementInitRequest, *handler.ParseError) {
+	dbCreateReq := new(admin.DatabaseCreateRequest)
+	if err := jsonpb.Unmarshal(r.Body, dbCreateReq); err != nil {
+		return nil, nil, handler.NewParseError(err, http.StatusBadRequest)
+	}
+
+	if dbCreateReq.Type != dbTypeLocal {
+		return nil, nil, handler.NewParseError(errInvalidDBType, http.StatusBadRequest)
+	}
+
+	if util.HasEmptyString(dbCreateReq.NamespaceName, dbCreateReq.Type) {
+		return nil, nil, handler.NewParseError(errMissingRequiredField, http.StatusBadRequest)
+	}
+
+	namespaceAddRequest := defaultedNamespaceAddRequest(dbCreateReq)
+	placementInitRequest := defaultedPlacementInitRequest(dbCreateReq)
+
+	return namespaceAddRequest, placementInitRequest, nil
+}
+
+func defaultedNamespaceAddRequest(r *admin.DatabaseCreateRequest) *admin.NamespaceAddRequest {
+	options := dbnamespace.NewOptions()
+
+	switch r.Type {
+	case dbTypeLocal:
+		options.SetBootstrapEnabled(true)
```

Comment:
afair, these are the default - any reason to explicitly set them here again?

Reply:
I thought it would be safer to not assume that we'd want the defaults for a local deployment compared to a clustered version, so explicitly setting them would be better. Tell me if you disagree.

Reply:
I agree in spirt but there's nuance to it. E.g. consider that Snapshotting is currently disabled by default, but as soon as we finish productionalising it, we'd turn it on by default in the namespace.NewOptions(). It'd be better to not expect us to remember to have to come here to enable it too. 

Instead of overriding all properties, I think you should override only specific properties you think explicitly need to be different in the local v clustered version. Sound fair?